Prefix assert_async with underscore, fix some bugs in assert_async CUDA testing

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53276 Prefix assert_async with underscore, fix some bugs in assert_async CUDA testing**

- One of the tests had a syntax error (but the test
  wasn't fine grained enough to catch this; any error
  was a pass)
- Doesn't work on ROCm

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26820048](https://our.internmc.facebook.com/intern/diff/D26820048)

ezyang committed:
Fix some bugs in assert_async CUDA testing

- One of the tests had a syntax error (but the test
  wasn't fine grained enough to catch this; any error
  was a pass)
- Doesn't work on ROCm

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Fix some bugs in assert_async CUDA testing"


- One of the tests had a syntax error (but the test
  wasn't fine grained enough to catch this; any error
  was a pass)
- Doesn't work on ROCm

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26820048](https://our.internmc.facebook.com/intern/diff/D26820048)

[ghstack-poisoned]

ezyang committed:
Update on "Fix some bugs in assert_async CUDA testing"


- One of the tests had a syntax error (but the test
  wasn't fine grained enough to catch this; any error
  was a pass)
- Doesn't work on ROCm

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26820048](https://our.internmc.facebook.com/intern/diff/D26820048)

[ghstack-poisoned]

mruberry reviewed (approved):
Stamped!

mruberry commented on torch/_torch_docs.py:
```diff
@@ -10565,9 +10565,9 @@ def merge_dicts(*dicts):
 """)
 
-add_docstr(torch.assert_async,
+add_docstr(torch._assert_async,
```
Since this isn't added to torch.rst it won't appear.

I don't think assert_async() was added to torch.rst, either?

ezyang replied:
No it wasn't. I guess it's fine for it to not show up now that it is underscored.

ezyang committed:
Update on "Prefix assert_async with underscore, fix some bugs in assert_async CUDA testing"


- One of the tests had a syntax error (but the test
  wasn't fine grained enough to catch this; any error
  was a pass)
- Doesn't work on ROCm

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26820048](https://our.internmc.facebook.com/intern/diff/D26820048)

[ghstack-poisoned]

ezyang committed:
Update on "Prefix assert_async with underscore, fix some bugs in assert_async CUDA testing"


- One of the tests had a syntax error (but the test
  wasn't fine grained enough to catch this; any error
  was a pass)
- Doesn't work on ROCm

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26820048](https://our.internmc.facebook.com/intern/diff/D26820048)

[ghstack-poisoned]
